packages/debug/src/browser/debug-session-connection.ts

```diff
@@ -78,7 +78,7 @@ export interface DebugEventTypes {
     'breakpoint': DebugProtocol.BreakpointEvent
     'capabilities': DebugProtocol.CapabilitiesEvent
     'continued': DebugProtocol.ContinuedEvent
-    'exited': DebugExitEvent
+    'exited': DebugProtocol.ExitedEvent,
```

Comment:
Why was this event changed? Using the DebugProtocol event loses the `reason` field exposed by `DebugExitEvent`.

If this change is needed, I believe the `DebugExitEvent` interface above is no longer used.

Reply:
Because there already is an "ExitedEvent" in the debug adapter protocol that is sent when the debug adapter learns that the debuggee exited. 
The `DebugExitEvent` is a fake event that was sent to signal that something went wrong with the connection to the debug adapter (not the debuggee). The decision to remove fake (or "generated" events) was made in order to make it simpler to understand the debug session lifecycle to make it more robust. If we need to communicate this error to the debug session, we need to establish a different channel, IMO.

Reply:
You're right that it's no longer used.

Reply:
OK, if this is being removed, please be aware it breaks an established API (we rely on detecting errors through this event).